locust/runners.py

```diff
@@ -816,6 +837,9 @@ def heartbeat_worker(self):
                     logger.info("Worker %s failed to send heartbeat, setting state to missing." % str(client.id))
                     client.state = STATE_MISSING
                     client.user_classes_count = {}
+                    if self._users_dispatcher is not None:
```

Comment:
it seems we're checking whether _user_dispatcher is not None in a lot of places. Is there a better way to handle this? Spawn the dispatcher earlier so that it always exists maybe? Idk, maybe a bad idea, but it looks a little ugly atm.

Reply:
Yes, I did those modifications quickly before my short vacation.

The reason this is like this is to handle cases where locust is stopped or is asked to quit when there's no load test running. When a load test is not running, `self._users_dispatcher` is `None`. Perhaps it's ugly, but with the manual testing I performed, it worked well.